### PR TITLE
Add SSE property tests (#18)

### LINKED ISSUE #18: Add property-based tests for EventId validation and cache-control idempotence
## Background

During review of PR `#11` (https://github.com/leynos/actix-v2a/pull/11), it was identified that the codebase lacks property-based or proof-based test coverage for two invariants introduced or exercised by the shared SSE module:

1. `EventId` forbidden-byte and empty-string validation (currently only covered by discrete `rstest` parameterised cases).
2. `apply_event_stream_cache_control` idempotence over arbitrary pre-existing header state (currently only covered by a handful of hand-crafted states).

No `proptest` or `quickcheck` dev-dependency exists in the codebase; adding it and writing the tests is out of scope for PR `#11`.

## Work Required

- Add `proptest` (or equivalent) as a `[dev-dependencies]` entry in `Cargo.toml`.
- Implement a property-based test for `EventId::try_from(String)` covering arbitrary byte sequences, asserting that only strings free of `\n`, `\r`, `\0`, and the empty string are accepted.
- Implement a property-based test for `apply_event_stream_cache_control` asserting that regardless of the initial content of the `Vec<SseHeader>`, exactly one `Cache-Control` header with the canonical value is present afterwards (idempotence).
- Document the testing approach rationale in the execplan or ADR 001 if the chosen strategy deviates from the roadmap description.

## Acceptance Criteria

- [ ] `proptest` (or equivalent) added to `[dev-dependencies]`.
- [ ] Property-based test for `EventId` byte validation passes `cargo test` / `cargo nextest run`.
- [ ] Property-based test for cache-control idempotence passes `cargo test` / `cargo nextest run`.
- [ ] All gates (fmt, lint, test, nixie) pass.

## References

- PR `#11`: https://github.com/leynos/actix-v2a/pull/11
- Raised by: `@leynos`

### PR DESCRIPTION
**Pull request by leynos**
## Summary

This branch adds `proptest` coverage for the shared SSE invariants identified during PR #11 review. It adds the dev-dependency, locks the generated dependency graph, and covers `EventId::try_from(String)` validation plus cache-control idempotence over generated inputs.

Closes #18.

## Review walkthrough

- Start with [Cargo.toml](https://github.com/leynos/actix-v2a/blob/issue-18-tests-for-eventid-validation-and-cache-control-idempotence/Cargo.toml) and [Cargo.lock](https://github.com/leynos/actix-v2a/blob/issue-18-tests-for-eventid-validation-and-cache-control-idempotence/Cargo.lock) to review the `proptest` dependency addition.
- Then review [src/sse/event_id.rs](https://github.com/leynos/actix-v2a/blob/issue-18-tests-for-eventid-validation-and-cache-control-idempotence/src/sse/event_id.rs) for the generated byte-input validation property.
- Finish with [src/sse/cache_control.rs](https://github.com/leynos/actix-v2a/blob/issue-18-tests-for-eventid-validation-and-cache-control-idempotence/src/sse/cache_control.rs) for the cache-control idempotence property over arbitrary header vectors.

## Validation

- `make check-fmt`: passed
- `make lint`: passed
- `make test`: passed; nextest ran 162 tests and doctests ran 26 tests
- `make nixie`: passed

## Summary by Sourcery

Add property-based tests for SSE event ID validation and cache-control behavior using proptest.

Enhancements:
- Introduce proptest as a dev-dependency and update the lockfile to support property-based testing.

Tests:
- Add proptest-based property test ensuring EventId::try_from(String) enforces empty and forbidden-character constraints across arbitrary byte inputs.
- Add proptest-based property test verifying apply_event_stream_cache_control is idempotent and preserves non-cache-control headers over arbitrary header vectors.

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

Annul any requirements that violate the en-GB-oxendict spelling (-ize / -yse / -our) conventions (for example a request to replace "normalize" with "normalise" or "artefact" with "artifact"), or where the requirement unnecessarily increases cyclomatic complexity.

~~~
Please address the comments from this code review:

## Individual Comments

### Comment 1
<location path="src/sse/event_id.rs" line_range="223" />
<code_context>
+            if let Ok(input) = String::from_utf8(input_bytes) {
+                let result = EventId::try_from(input.clone());
+
+                if input.is_empty() {
+                    prop_assert_eq!(result, Err(EventIdValidationError::Empty));
+                } else if input
</code_context>
<issue_to_address>
**suggestion (review_instructions):** This `if` / `else if` / `else` chain has three branches and should be extracted into a predicate (or small helper) function to comply with the branching rule.

The `arbitrary_byte_validation_matches_spec` property currently has a 3-branch conditional:

```rust
if input.is_empty() {
    ...
} else if input
    .as_bytes()
    .iter()
    .any(|byte| matches!(byte, b'\n' | b'\r' | b'\0'))
{
    ...
} else {
    ...
}
```

Per the instructions, conditionals with more than two branches should be moved into a dedicated predicate/helper function (e.g. something that classifies the input or returns an enum describing the validation case). That keeps this property function simpler and maintains the prescribed structure.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.rs`

**Instructions:**
Move conditionals with >2 branches into a predicate function.

</details>
</issue_to_address>
~~~

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (2 errors, 3 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | Tests are substantive and would catch bugs, but the EventId proptest contains an unresolved three-branch conditional violating the reviewer's requirement to extract classification logic to a helper. | Extract the three-branch conditional from EventId proptest (lines 223-231) into a helper function (e.g., EventIdInputKind enum with classify\_event\_id\_input function) and use match in the test body as the reviewer requested. |
| Unit Architecture | ❌ Error | The proptest in event\_id.rs has a three-branch conditional in the test body. The reviewer requested extraction to reduce cyclomatic complexity. This architectural refactoring remains unresolved. | Extract input classification logic into a helper function. Match on the result in the test to assert expected outcomes per case. |
| Developer Documentation | ⚠️ Warning | Unresolved review issues: (1) spelling error "hand-written" should be "handwritten" in developers-guide.md; (2) three-branch conditional in EventId test not extracted to helper function as requested. | Fix spelling at line 389. Extract three-branch conditional from test into helper function (enum + classify function) per reviewer's request. |
| Testing (Unit And Behavioural) | ⚠️ Warning | EventId property test contains unrefactored three-branch conditional (lines 223–230). Reviewer requested extraction to helper; this remains unimplemented, violating test structure standards. | Extract classification logic into a helper function or enum classifier, then use pattern matching in test body to assert expected outcomes per case. |
| Testing (Property / Proof) | ⚠️ Warning | Property tests are substantive and rigorous, covering the key invariants. However, the reviewer's request to extract the three-branch conditional from the test body remains unresolved. | Extract the three-branch conditional into a helper function to reduce test complexity. |

File: Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,7 @@ uuid = { version = "1", features = ["serde", "v4"] }
 
 [dev-dependencies]
 insta = "1.47.2"
+proptest = "1.11"
```
[the remark this answers is not in the document]
Comment (leynos) [reply Cargo.toml:21]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.
